genesis/engine/solvers/rigid/rigid_validate.py

```diff
@@ -0,0 +1,65 @@
+import taichi as ti
+
+import genesis.utils.array_class as array_class
+
+from genesis.engine.solvers.rigid.rigid_debug import Debug
+from .contact_island import INVALID_NEXT_HIBERNATED_ENTITY_IDX
+
+
+@ti.func
+def validate_entity_hibernation_state_for_all_entities_in_temp_island(
```

Comment:
@duburcqa do you think we should introduce those Debug and assert in our kernel code?

Reply:
> do you think we should introduce those Debug and assert in our kernel code?

I'm not against it if it works transparently and easy to maintain support for gstaichi dev team (basically Hugh for now).

Reply:
@YilingQiao can you define what you mean by "our kernel code" please. I interpreted this to mean "our Genesis GPU kernels", but alexis seems to be interpreting it to mean "support natively within gstaichi".

Reply:
I mean `our Genesis GPU kernels`.